homeassistant/components/dialogflow/__init__.py

```diff
@@ -38,52 +31,72 @@ class DialogFlowError(HomeAssistantError):
 
 async def async_setup(hass, config):
     """Set up Dialogflow component."""
-    hass.http.register_view(DialogflowIntentsView)
-
     return True
 
 
-class DialogflowIntentsView(HomeAssistantView):
-    """Handle Dialogflow requests."""
-
-    url = INTENTS_API_ENDPOINT
-    name = 'api:dialogflow'
-
-    async def post(self, request):
-        """Handle Dialogflow."""
-        hass = request.app['hass']
-        message = await request.json()
-
-        _LOGGER.debug("Received Dialogflow request: %s", message)
-
-        try:
-            response = await async_handle_message(hass, message)
-            return b'' if response is None else self.json(response)
-
-        except DialogFlowError as err:
-            _LOGGER.warning(str(err))
-            return self.json(dialogflow_error_response(
-                hass, message, str(err)))
+async def handle_webhook(hass, webhook_id, request):
+    """Handle incoming webhook with Dialogflow requests."""
+    message = await request.json()
+
+    _LOGGER.debug("Received Dialogflow request: %s", message)
+
+    try:
+        response = await async_handle_message(hass, message)
+        return b'' if response is None else HomeAssistantView.json(response)
```

Comment:
I don't think that it's a webhook when the caller expects a response.

Reply:
Forget about the comment above.

However, we should not use `HomeAssistantView` but instead use aiohttp directly.

```python
from aiohttp import web

return web.json_response(…)
```

[Docs](https://docs.aiohttp.org/en/stable/web_reference.html#json-response)